packages/contracts/contracts/L1/AssetManager.sol

```diff
@@ -0,0 +1,1428 @@
+        uint128 totalPendingKroShares;
+        uint128 totalPendingKghShares;
+    }
+
+    /**
+     * @notice Constructs the reward information of the vault of a validator.
+     *
+     * @custom:field boostedReward                 Cumulated boosted reward for KGH delegators in the vault.
+     * @custom:field validatorRewardKro            Cumulated reward for the validator.
+     * @custom:field commissionRateNumerator       Numerator of the commission rate.
+     * @custom:field commissionRateMaxChangeRange  Range that a commission rate can be changed to.
+     * @custom:field lastCommissionRateChangedTime Last timestamp when the commission rate was changed.
+     */
+    struct Reward {
+        uint128 boostedReward;
+        uint128 validatorRewardKro;
+        uint8 commissionRateNumerator;
+        uint8 commissionRateMaxChangeRange;
+        uint128 lastCommissionRateChangedTime;
+    }
+
+    /**
+     * @notice Constructs the delegator of KRO in the vault of a validator.
+     *
+     * @custom:field shares                 Amount of shares for KRO delegation.
+     * @custom:field undelegateRequestTimes Timestamps of undelegation requests.
+     * @custom:field undelegateRequests     Pending shares at timestamps undelegations are requested.
+     */
+    struct KroDelegator {
+        uint128 shares;
+        uint256[] undelegateRequestTimes;
+        mapping(uint256 => uint128) undelegateRequests;
+    }
+
+    /**
+     * @notice Constructs the delegator of KGH in the vault of a validator.
+     *
+     * @custom:field kroShares              Amount of shares for KRO delegation.
+     * @custom:field kghShares              Amount of shares for delegation of KGH with a corresponding tokenId.
+     * @custom:field undelegateRequestTimes Timestamps of undelegation requests.
+     * @custom:field undelegateRequestedIds TokenIds of KGHs for undelegation at timestamps undelegations
+     *                                      are requested.
+     * @custom:field undelegateRequests     Pending KGH shares for undelegation at timestamps undelegations
+     *                                      are requested.
+     */
+    struct KghDelegator {
+        mapping(uint256 => uint128) kroShares;
+        mapping(uint256 => uint128) kghShares;
+        uint256[] undelegateRequestTimes;
+        mapping(uint256 => uint256[]) undelegateRequestedIds;
+        mapping(uint256 => uint128) undelegateRequests;
+    }
+
+    /**
+     * @notice Constructs the vault of a validator.
+     *
+     * @custom:field asset             Asset information of the vault.
+     * @custom:field reward            Reward information of the vault.
+     * @custom:field isInitiated       Whether the vault is initiated.
+     * @custom:field noSubmissionCount Number of counts that the validator did not submit the output in priority round.
+     */
+    struct Vault {
+        Asset asset;
+        Reward reward;
+        bool isInitiated;
```

Comment:
This field is not used anywhere in `AssetManager` contract, but it will be used in `ValidatorManager`?
And why don't rename to `isActivated`?

Reply:
Yep it will be used only in `ValidatorManager`. How do you think about this suggestion @seolaoh ?

Reply:
Yes it will be used in `ValidatorManager` and validator can be initiated but in `inactive` status, so the name is appropriate to separate the status. Let's check it in the ValidatorManager PR.

Reply:
OK!